da65: adding support for 4510 cpu of c65

SvOlli opened the pull request:
This time I've added support for the 4510 cpu to the disassembler da65.

oliverschmidt commented on src/da65/handler.c:
```diff
@@ -508,9 +546,24 @@ void OH_DirectIndirectLongX (const OpcDesc* D attribute ((unused)))
 
 
 
+static void impl_StackRelativeIndirectY (const char *sp, const OpcDesc* D attribute ((unused)))
```
For the sake of consistency with the rest of this file I'd prefer to have this function removed and the OneLine() directly put into the two OH_...() functions.

SvOlli replied:
Good catch. The original impl_* function was much longer, but got shortened during development, so that I didn't recognize the redundancy in there.

oliverschmidt commented on src/da65/opc4510.c:
```diff
@@ -0,0 +1,306 @@
+    {   "bbr7", 3,  flUseLabel,               OH_BitBranch             }, /* $7f */
+    {   "bra",  2,  flLabel,                  OH_Relative              }, /* $80 */
+    {   "sta",  2,  flUseLabel,               OH_DirectXIndirect       }, /* $81 */
+    {   "sta",  2,  flNone,                   OH_StackRelativeIndirectY4510}, /* $82 */
```
Please adjust the formatting of the rest of the table to align with this line.

oliverschmidt commented:
@greg-king5: Do you have comments on this pull request?

SvOlli commented:
Additional comments on this pull request:
- da65 now has support for 4510 cpu
- added directory test/disassembler for simple da65 testcases, similar to testcases in test/assembler
- implemented testcase for 4510 cpu only
- da65 documentation updated for 4510 support
(Also remove static function and fixed table formatting)

greg-king5 commented on doc/da65.sgml:
```diff
@@ -239,6 +241,11 @@ disassembler may be told to recognize either the 65SC02 or 65C02 CPUs. The
 
+When disassembling 4510 code, due to handling of 16-bit wide branches, da65
+can produce output that can not be re-assembled, when one or more of those
+branches point outside of the disassmbled memory. This can happen when text
```
Typo: disassmbled -> disassembled

greg-king5 commented on test/disassembler/Makefile:
```diff
@@ -0,0 +1,41 @@
+	@$$(BINDIR)/da65 --cpu $(1) $(START) -o $$@ $$<
+
+$$(WORKDIR)/$(1)-reass.bin: $$(WORKDIR)/$(1)-reass.s
+	@$$(BINDIR)/cl65 --cpu $(1) -t none $(START) --obj-path $$(WORKDIR) -o $$@ $$<
```
Add a `@rm` command for that second assembly.
